eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -638,4 +661,21 @@ private static SimpleModule module() {
     module.addDeserializer(AssignmentTestCase.class, new AssignmentTestCaseDeserializer());
     return module;
   }
+
+  private static void setBaseClientHttpClientOverrideField(EppoHttpClient httpClient) {
+    setBaseClientOverrideField("httpClientOverride", httpClient);
+  }
+
+  /** Uses reflection to set a static override field used for tests (e.g., httpClientOverride) */
+  @SuppressWarnings("SameParameterValue")
+  public static <T> void setBaseClientOverrideField(String fieldName, T override) {
+    try {
+      Field httpClientOverrideField = BaseEppoClient.class.getDeclaredField(fieldName);
+      httpClientOverrideField.setAccessible(true);
+      httpClientOverrideField.set(null, override);
+      httpClientOverrideField.setAccessible(false);
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
why do we need this reflection again? if we don't want to expose this publicly, we can just make it package-private